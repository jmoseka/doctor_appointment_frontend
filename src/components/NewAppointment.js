import React from 'react'

const NewAppointment = () => {
  return (
    <div>NewAppointment</div>
  )
}

export default NewAppointment