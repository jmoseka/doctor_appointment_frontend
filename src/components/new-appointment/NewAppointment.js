import React, { useEffect, useState } from 'react';
import './new_Appointment.css';
import { useDispatch, useSelector } from 'react-redux';
import { reserveAction } from '../../redux/my_appointment_redux/my_appointment';
import { getDoctorsAction } from '../../redux/doctorReduce/doctors';

const NewAppointment = () => {
  const dispatch = useDispatch();
  // const reserveDoctor = useSelector((state) => state.my_appointment);
  const doctors = useSelector((state) => state.doctor);
  const userId = JSON.parse(localStorage.getItem('user'));

  const cityTown = document.getElementById('city-town');

  const [doctorId, setDoctorId] = useState('');
  const [city, setCity] = useState('');
  const [dateReserved, setDateReserved] = useState('');

  useEffect(() => {
    dispatch(getDoctorsAction());
  }, [dispatch]);

  const handleSubmit = (e) => {
    e.preventDefault();
    if (doctorId !== '' && dateReserved !== '') {
      const data = {
        user_id: userId.user.id,
        city,
        date_reserved: dateReserved,
        doctor_id: doctorId,
      };
      cityTown.value = '';
      dispatch(reserveAction(data));
    }
  };

  return (
    <div className="reserveContainer">
      <div>
        <div className="flex">
          <h4 className="reserveFormHeading">Book Appointment</h4>
        </div>

        <div className="resrve-header">
          <p className="text-doctors">
            We have different doctors from different parts of the world.
            <br />
            If you wish to find out if we have a doctor who is available in your
            city, please use the selector below
          </p>
        </div>

        <form action="/#" className="form-list" onSubmit={handleSubmit}>
          <div className="details-form">
            <div className="form-group">
              <p className="ms-4 mb-1 text-white">City</p>
              <input
                type="text"
                name="city"
                id="city-town"
                className="form-control"
                placeholder="Enter your city"
                required
                onChange={(e) => setCity(e.target.value)}
              />
            </div>
            <div className="form-group">
              <p className="ms-4 mb-1 text-white">Doctor</p>
              <select
                name="availableDoctors"
                id="availableDoctors"
                value={doctorId}
                className="form-control"
                onChange={(e) => setDoctorId(e.target.value)}
              >
                {doctors.map((doctor) => (
                  <option value={doctor.id} key={doctor.id}>
                    {doctor.name}
                  </option>
                ))}
              </select>
            </div>
            <div className="form-group">
              <p className="ms-4 mb-1 text-white">Date</p>
              <input
                type="date"
                value={dateReserved}
                className="form-control"
                onChange={(e) => setDateReserved(e.target.value)}
              />
            </div>
          </div>
          <div className="form-book mb-3">
            <input
              type="submit"
              value="Book Now"
              className="form-user-button"
            />
          </div>
        </form>
      </div>
    </div>
  );
};
export default NewAppointment;
