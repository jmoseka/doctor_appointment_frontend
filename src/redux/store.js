import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { SignUpReducer } from './auth/register';
import { LoginUserReducer } from './auth/login';
import myAppointmentReducer from './my_appointment_redux/my_appointment';

const reducer = combineReducers({
  user: SignUpReducer,
  login: LoginUserReducer,
  my_appointment: myAppointmentReducer,
});

const store = createStore(
  reducer, applyMiddleware(logger, thunk),
);

export default store;
