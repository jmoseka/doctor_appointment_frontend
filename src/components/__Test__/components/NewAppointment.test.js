import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/store';
import NewAppointment from '../../new-appointment/NewAppointment';

describe('BookAppointment', () => {
  it('renders correctly', () => {
    window.localStorage.setItem('user', 1);
    const tree = render(
      <Provider store={store}>
        <Router>
          <NewAppointment />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
