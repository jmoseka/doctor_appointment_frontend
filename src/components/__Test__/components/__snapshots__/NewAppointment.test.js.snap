// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookAppointment renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="reserveContainer"
      >
        <div>
          <div
            class="flex"
          >
            <h4
              class="reserveFormHeading"
            >
              Book Appointment
            </h4>
          </div>
          <div
            class="resrve-header"
          >
            <p
              class="text-doctors"
            >
              We have different doctors from different parts of the world.
              <br />
              If you wish to find out if we have a doctor who is available in your city, please use the selector below
            </p>
          </div>
          <form
            action="/#"
            class="form-list"
          >
            <div
              class="details-form"
            >
              <div
                class="form-group"
              >
                <p
                  class="ms-4 mb-1 text-white"
                >
                  City
                </p>
                <input
                  class="form-control"
                  id="city-town"
                  name="city"
                  placeholder="Enter your city"
                  required=""
                  type="text"
                />
              </div>
              <div
                class="form-group"
              >
                <p
                  class="ms-4 mb-1 text-white"
                >
                  Doctor
                </p>
                <select
                  class="form-control"
                  id="availableDoctors"
                  name="availableDoctors"
                />
              </div>
              <div
                class="form-group"
              >
                <p
                  class="ms-4 mb-1 text-white"
                >
                  Date
                </p>
                <input
                  class="form-control"
                  type="date"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-book mb-3"
            >
              <input
                class="form-user-button"
                type="submit"
                value="Book Now"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="reserveContainer"
    >
      <div>
        <div
          class="flex"
        >
          <h4
            class="reserveFormHeading"
          >
            Book Appointment
          </h4>
        </div>
        <div
          class="resrve-header"
        >
          <p
            class="text-doctors"
          >
            We have different doctors from different parts of the world.
            <br />
            If you wish to find out if we have a doctor who is available in your city, please use the selector below
          </p>
        </div>
        <form
          action="/#"
          class="form-list"
        >
          <div
            class="details-form"
          >
            <div
              class="form-group"
            >
              <p
                class="ms-4 mb-1 text-white"
              >
                City
              </p>
              <input
                class="form-control"
                id="city-town"
                name="city"
                placeholder="Enter your city"
                required=""
                type="text"
              />
            </div>
            <div
              class="form-group"
            >
              <p
                class="ms-4 mb-1 text-white"
              >
                Doctor
              </p>
              <select
                class="form-control"
                id="availableDoctors"
                name="availableDoctors"
              />
            </div>
            <div
              class="form-group"
            >
              <p
                class="ms-4 mb-1 text-white"
              >
                Date
              </p>
              <input
                class="form-control"
                type="date"
                value=""
              />
            </div>
          </div>
          <div
            class="form-book mb-3"
          >
            <input
              class="form-user-button"
              type="submit"
              value="Book Now"
            />
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
